test/Ocelot.AcceptanceTests/ClientRateLimitTests.cs

```diff
@@ -0,0 +1,208 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Ocelot.Configuration.File;
+using Shouldly;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using TestStack.BDDfy;
+using Xunit;
+
+namespace Ocelot.AcceptanceTests
+{
+    public class ClientRateLimitTests : IDisposable
```

Comment:
I think we need one more acceptance test that checks if rate limiting is set the first request is not accidentally rate limited. The test should_call_withratelimiting() only checks the last reponse was rate limited. Perhaps we should check each response to make sure it is what we expect.